causalpy/pymc_models.py

```diff
@@ -39,12 +57,59 @@ def build_model(self, X, y, coords) -> None:
         raise NotImplementedError("This method must be implemented by a subclass")
 
     def _data_setter(self, X) -> None:
+        """
+        Set data for the model.
+
+        This method is used internally to register new data for the model for
+        prediction.
+        """
         with self.model:
             pm.set_data({"X": X})
 
     def fit(self, X, y, coords: Optional[Dict[str, Any]] = None) -> None:
-        """Draw samples from posterior, prior predictive, and posterior predictive
-        distributions.
+        """Draw samples fromposterior, prior predictive, and posterior predictive
+        distributions, placing them in the model's idata attribute.
+
+        .. note::
+            Calls the build_model method
+
+        Example
+        -------
+        >>> import numpy as np
+        >>> import pymc as pm
+        >>> from causalpy.pymc_models import ModelBuilder
+        >>> class MyToyModel(ModelBuilder):
+        ...    def build_model(self, X, y, coords):
+        ...        with self:
+        ...            X_ = pm.MutableData(name="X", value=X)
+        ...            y_ = pm.MutableData(name="y", value=y)
+        ...            beta = pm.Normal("beta", mu=0, sigma=1, shape=X_.shape[1])
+        ...            sigma = pm.HalfNormal("sigma", sigma=1)
+        ...            mu = pm.Deterministic("mu", pm.math.dot(X_, beta))
+        ...            pm.Normal("y_hat", mu=mu, sigma=sigma, observed=y_)
+        >>> rng = np.random.default_rng(seed=42)
+        >>> X = rng.normal(loc=0, scale=1, size=(20, 2))
+        >>> y = rng.normal(loc=0, scale=1, size=(20,))
+        >>> model = MyToyModel(sample_kwargs={"chains": 2, "draws": 2})
+        >>> model.fit(X, y)
+        Only 2 samples in chain.
+        Auto-assigning NUTS sampler...
+        Initializing NUTS using jitter+adapt_diag...
+        Multiprocess sampling (2 chains in 4 jobs)
+        NUTS: [beta, sigma]
+        Sampling 2 chains for 1_000 tune and 2 draw iterations (2_000 + 4 draws total)
+          took 0 seconds.gences]
+        The number of samples is too small to check convergence reliably.
+        Sampling: [beta, sigma, y_hat]
+        Sampling: [y_hat]
+        Inference data with groups:
+                > posterior
+                > posterior_predictive
+                > sample_stats
+                > prior
+                > prior_predictive
+                > observed_data
+                > constant_data
```

Comment:
So I am not 100% sure about this, but I think one possible good idea would be:
* Not include the text output in the docstring examples. My feeling is that this adds clutter and could make things harder to keep up to date.
* The possible exception to this would be for specific methods, such as `result.summary()` whose sole job is to provide text output.

I'll invite thoughts from @lucianopaz and @juanitorduz on this.

Reply:
I like this idea and agree it would reduce clutter and be easier to maintain.

Reply:
I also agree with Ben's comment :)